### PR TITLE
Added method flatten()

### LINKED ISSUE #15: New Method: flatten()

### PR DESCRIPTION
**Pull request by islas27**
- Added new method flatten() to Array.prototype
- Added the unit test
- Added the documentation & examples
- Added new test data

Closes #15

### CODE REVIEW

File: test/flatten.spec.js [hunk 1]
```diff
@@ -0,0 +1,22 @@
...
+  })
+
+  it('returns the same array when no flatten is needed', () => {
+    expect(data.numbers.flatten()).to.eql(data.numbers)
+    expect(data.letters.flatten()).to.eql(data.letters)
+    expect(['a']).to.eql(['a'])
+  })
+
+  it('returns the correct answer', () => {
+    expect(data.unflattenedArrayNumbersS.flatten()).to.eql(data.flattenedArrayNumbersS)
```
Comment (ryanez) [inline test/flatten.spec.js:17]: Please add a test to validate your method is returning a new array and it's not modifying the original one.

**Comment by islas27** [comment]
Added the unit test and fixed a bit the code, its true it was modifying the original array. Used `slice()` to get a copy and then modify that one with `splice()`. Check commit 0ede140

File: aee.js [hunk 2]
```diff
@@ -141,3 +141,15 @@ Array.prototype.min = function (comparer) {
...
+Array.prototype.flatten = function () {
+  let copy = this.slice()
+  let len = copy.length
+  for (let i = 0; i < len; i += 1) {
+    if (Array.isArray(copy[i])) {
+      len += copy[i].length - 1
+      copy.splice(i, 1, ...copy[i])
+    }
+  }
+  return copy
```
Comment (ryanez) [inline aee.js]: If you take a look at the whole file this `copy` guy is the black in the rice, I think all the `result` boys will bully him.

Comment (islas27) [reply aee.js]: yeah, it seems it was too late for me to keep working... hahaha, I'm on it!

**Comment by ryanez** [comment]
LGTM 👍

**Pull request merged**